### DISCUSSION TITLE
Tokyonight transparency recipe and fzf-lua

### DISCUSSION THREAD

**Discussion by pkazmier** [LazyVim/LazyVim#5239] [Q&A] [answered]
Has anyone figured out how to make fzf-lua transparent? I've followed the [Tokyonight transparent recipe](https://www.lazyvim.org/configuration/recipes#make-tokyonight-transparent) in the LazyVim documentation, which works well for the most part, but the fzf-lua picker is not transparent (although the preview is). At first, I thought it might be the embedded terminal it runs in, but that is not the case as you can see the terminal opened at the bottom of my screen.

<img width="1865" alt="image" src="https://github.com/user-attachments/assets/006d0a32-5bf1-49ba-80c3-9f09f7b81273" />

**Comment by ibhagwan** [comment, marked as answer]
For transparency you may need to set fzf’s bg to -1, see https://github.com/ibhagwan/fzf-lua/discussions/1580#discussioncomment-11607126

**Reply by pkazmier** [reply]
Ah, yes, that did the trick. Thank you! 
<img width="2002" alt="image" src="https://github.com/user-attachments/assets/124998c6-8616-433b-bce1-e77b865f1e6f" />

I'll submit a PR for the LazyVim doc on using transparency with Tokyonight to reflect this now that fzf-lua is the default in LazyVim. Oh, and by the way, I love fzf-lua. Nice work!

**Comment by ibhagwan** [comment]
> I'll submit a PR for the LazyVim doc on using transparency with Tokyonight to reflect this now that fzf-lua is the default in LazyVim. Oh, and by the way, I love fzf-lua. Nice work!

Ty for the kind words, a bit of explanation why this happens for your PR, by using `fzf_colors=true` the fzf terminal colorscheme is auto generated and links `FzfLuaFzfNormal` to neovim’s `Normal` which contains both fg and bg colors, those in turn are set using fzf’s `--color` argument as fg and bg respectively, while that works for most use cases, as you found out it won’t work for transparency, for that the color spec needs to be `-1` which is only possible via `fzf_colors`:

From `man fzf`:
```
              ANSI COLORS:
                  -1         Default terminal foreground/background color
                             (or the original color of the text)
                  0 ~ 15     16 base colors
```

**Reply by pkazmier** [reply]
Thank you for the explanation!
